### ISSUE TITLE
Add QS gazetteer points for places not already in WOF

### ISSUE THREAD

**Issue by nvkelso** [whosonfirst-data/whosonfirst-data#107]
1,300,000 total features (yikes!):

- [x] 72k of  have been imported from QS point data from _before, before_
- [x] 465k (remainder of the 518k - 72k) to parse thru for the gn_then_gp_w_content file
    - https://github.com/whosonfirst-data/whosonfirst-data/pull/824
- [ ] 800k left over in the big honking WOF point gazetteer file
    - Requires some wrangling by @nvkelso to extract the `qs_id` onto those files

**Comment by stepps00** [comment]
Similar to Issue #222.

**Comment by stepps00** [comment]
A rundown of files in the S3 misc bucket:

[`import.geojson`](https://s3.amazonaws.com/whosonfirst.mapzen.com/misc/Issue%20107/import.geojson)
* All features in this file are ready for import.
* Contains a mix of placetypes and spans several countries. Disregard the `aaroncc`, `actionplan`,`_placetype`, and `_newplan` columns. All other attribute columns come directly from the Quattroshapes gazetteer points file. 
* 54,928 features

[`noimport.geojson`](https://s3.amazonaws.com/whosonfirst.mapzen.com/misc/Issue%20107/noimport.geojson)
* All features in this file should not be imported at this time.
* Who's on First already knows about the majority of these places, but we may import additional features from this file in the future.
* 449,777 features

[`additional__noimport_locality.geojson`](https://s3.amazonaws.com/whosonfirst.mapzen.com/misc/Issue%20107/additional_noimport_locality.geojson)
* All features in this file should not be imported at this time.
* These features were imported to Who's on First in March 2016. Contains a `wof:id` field that these features _should_ map to.
* 12,954 features

[`additional__noimport_localadmin.geojson`](https://s3.amazonaws.com/whosonfirst.mapzen.com/misc/Issue%20107/additional_noimport_localadmin.geojson)
* All features in this file should not be imported at this time.
* These features were imported to Who's on First in March 2016. Contains a `wof:id` field that these features _should_ map to.
* 356 features

[`additional_noimport_neighbourhood.geojson`](https://s3.amazonaws.com/whosonfirst.mapzen.com/misc/Issue%20107/additional_noimport_neighbourhood.geojson)
* All features in this file should not be imported at this time.
* These features were imported to Who's on First in March 2016. Contains a `wof:id` field that these features _should_ map to.
* 742 features

**Comment by stepps00** [comment]
New files in the S3 bucket (do not include geometries, yet):

[`import_w_wiki.geojson`](https://s3.amazonaws.com/whosonfirst.mapzen.com/misc/Issue%20107/import_w_wiki.geojson)
* This is the `import.geojson` file (from previous comment) _with_ wikipedia attributes appended.
* Wikipedia attributes have a prefix of `wk:`.

[`languages_QS_clean.json`](https://s3.amazonaws.com/whosonfirst.mapzen.com/misc/Issue%20107/languages_QS_clean.json)
* This .json file contains a `qs_id` field and all wikipedia language fields that associate to that `qs_id`.

Thanks @okavvada!

**Comment by nvkelso** [comment]
Related: https://github.com/whosonfirst-data/whosonfirst-data/issues/376.

**Comment by stepps00** [comment]
List of features in `noimport.geojson` per "placetype" note, with links to commit (complete if checked):

- [ ] country
- [ ] county
- [ ] county (deprecated)
- [ ] dependency
- [ ] localadmin
- [ ] locality
- [ ] locality (funky)
- [ ] locality (geoplanet) / neighbourhood (geonames)
- [ ] neighbourhood
- [ ] region
- [ ] region (deprecated)

**Comment by stepps00** [comment]
import plan notes:

```
GOAL
Import all of the QS Point Gazetteer into WOF
Current status:
- 72k of 1.3mm have been imported from QS point data
    - Which leaves 450k out of 518k to parse thru for the gn_then_gp_w_content file
- There are 465k features of mixed source, most of which are from the QS polygon data
        208221  locality    import if new, else join attr
        115494  localadmin  if a new county, then mark is_current=0; USA, AUS, BRA, SPA, UK, FRA, ITA, SWITZERLAND, KOREA, NETH, BELGIUM?
        90284   neigbourhood    import unless in curated city? With buffer?
        43726   county  if a new county, then mark is_current=0 (unless Algeria, Namibia, Botswana, Turkmin, Uzbek)
        4878    region  if a new region, then mark is_current=0
        1359    microhood   n/a
        373 macrocounty n/a
        252 macrohood   n/a
        224 country don't import
        107 borough n/a
        84  macroregion n/a
        40  disputed    n/a
        39  dependency  n/a
        11  empire  n/a
        465092      total
- there are 1.355k total QS point gazetteer files
    * Nathaniel to join these up!
    gn_then_gp - missing qs_id
    gp_then_gn - has     qs_id
- Remainder: 800k
    no-content
for the 450k:
- placetype????
    * MOD to read from data, not CLI argument
    0. investigate _placetype, can we just use this? YES but with funky caveats?
    1. placetype from GP for many but not all features
    2. if that doesn't exist, then use the lookup geonames_org_placetype
    3. else pretend like it's a locality
         locality @ 237195
         locality_gp_neighbourhood_gn @ 397 (import as locality)
         locality_funky @ 2461
         localadmin @ 115890
         neighbourhood @ 69243
         county @ 20242
         county_deprecated @ 9    (mz:is_current = 0 & edtf:cessation:2017-05-15)
         region @ 4072
         region_deprecated @ 37    (mz:is_current = 0 & edtf:cessation:2017-05-15)
         country @ 237
         dependency @ 32
         other @ n/a
- country code????
    * MOD to read from data, not CLI argument
    aaroncc
    gn_country
    name_adm0 lookup table to be created
    for the 7 that don't have one, go figure it out
        asia
        earth
        south america
        island
        mall
        random
        random
- country_id
    * lookup table from country code to wof:id from metafile
- parent_id: -1
- repo: whosonfirst-data
- fix the API client thing
- source: qs for prefix, (and quattroshapes)
- ignore ascii not ascii wof:name problems (there an issue to followup already)
- write out debug file (1 file per wof:country or wof:placetype)
BUILD PROPERTIES
- building up any concordances
    gn_id  = gn:id
    gp/woe = woe:id
    qs_id  = qs:id
- bring in any woe properties
    woe_adm0    = woe:adm0_id
    placetype   = woe:placetype
    name_adm0   = woe:name_adm0
    name_adm1   = woe:name_adm1
- bring in any gn properties
    gn_pop     = gn:pop
    gn_fcode   = gn:fcode
    gn_country = gn:country
- bring in any remaining qs properties
    name        = qs:name
    photos      = qs:photos
    photos_all  = qs:photos_all
    photos_1k   = qs:photos_1k
    photos_9k   = qs:photos_9k
    photos_sr   = qs:photos_sr
    photos_9r   = qs:photos_9r
    pop_sr      = qs:pop_sr
- special wof properties
    name        = wof:name
    placetype   = _placetype (but there are funky things here)
    gn_pop      = wof:population
    pop_sr      = wof:population_rank
    parent_id   = -1
    wof:country = &country logic, see above
    iso:country = &country logic, see above
    created     = unix timestamp
    - only if not present / not src:mz
        lbl:latitude  = geom.lat
        lbl:longitude = geom.long
- other
    src:geom         = quattroshapes
    src:lbl:centroid = quattroshapes
    src:population   = qs
    repo             = whosonfirst-data
match to existing feature, if available
    store alt geoms
```

**Comment by stepps00** [comment]
The GeoNames locality records have been merged through #108, which likely included any records we would gain from the leftover QS points. We can add concordances, but I'd suggest holding off on importing any new records.

Thoughts on closing this issue, @nvkelso?

**Comment by nvkelso** [comment]
Yeah, I'm fine closing.
